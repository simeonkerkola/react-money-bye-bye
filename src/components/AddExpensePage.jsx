import React from 'react'

const AddExpensePage = () => (
  <div>My add expense component</div>
)

export default AddExpensePage
