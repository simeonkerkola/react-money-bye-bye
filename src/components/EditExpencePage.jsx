import React from 'react'

const EditExpencePage = (props) =>  {
  console.log(props);
  return (
    <div>Edit Expense with the id of {props.match.params.id}</div>
  )
}
export default EditExpencePage
