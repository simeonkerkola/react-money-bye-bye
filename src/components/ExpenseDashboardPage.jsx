import React from 'react'

const ExpenseDashboardPage = (props) => (
  <div>
    Dashboard component
  </div>
)

export default ExpenseDashboardPage
