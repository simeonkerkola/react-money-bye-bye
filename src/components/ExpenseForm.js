import React from 'react'

export default class ExpenseForm extends React.Component {
  state = {
    description: '',
    note: '',
    amount: '',
  }

  onDescriptionChange = e => {
    const description = e.target.value
    this.setState(() => ({ description }))
  }
  onNoteChange = e => {
    const note = e.target.value
    this.setState(() => ({ note }))
  }
  onAmountChange = e => {
    const amount = e.target.value
    if (amount.match(/^\d*(\.\d{0,2})?$/)) {
        this.setState(() => ({ amount }))
    }
}
  render() {
    return (
      <div>
        <form action="">
          <input
            type="text"
            placeholder="Description"
            value={this.state.description}
            onChange={this.onDescriptionChange}
            autoFocus
          />
          <input
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <textarea
            cols="30"
            rows="4"
            placeholder="Add a note"
            value={this.state.note}
            onChange={this.onNoteChange}
          />
          <button>Add Expense</button>
        </form>
      </div>
    )
  }
}
