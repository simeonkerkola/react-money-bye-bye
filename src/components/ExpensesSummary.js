import React from 'react'
import { connect } from 'react-redux'
import selectExpensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
  return (
    <div>
      <h1>
        {expenseCount} {expenseWord} totalling {expensesTotal}â‚¬
      </h1>
    </div>
  )
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses).toFixed(2),
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
