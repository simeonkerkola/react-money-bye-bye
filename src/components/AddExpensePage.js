import React from 'react'
import ExpenseForm from './ExpenseForm.js'

const AddExpensePage = () => (
  <div>
    My add expense component
    <h1>Add Expense</h1>
    <ExpenseForm />
  </div>
)
export default AddExpensePage
