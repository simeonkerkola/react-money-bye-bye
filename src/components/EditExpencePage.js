import React from 'react'
import { connect } from 'react-redux'
import { editExpense, removeExpense } from '../actions/expenses'
import ExpenseForm from './ExpenseForm'

const EditExpencePage = props => (
  <div>
    <ExpenseForm
      expense={props.expense}
      onSubmit={(expense) => {
        props.dispatch(editExpense(props.expense.id, expense))
        props.history.push('/') // throw user to dashboard page
      }}
    />
    <button
      onClick={() => {
        props.dispatch(removeExpense({ id: props.expense.id }))
        props.history.push('/')
        console.log('click')
      }}
    >
      Remove
    </button>
  </div>
)

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id),
})

export default connect(mapStateToProps)(EditExpencePage)
