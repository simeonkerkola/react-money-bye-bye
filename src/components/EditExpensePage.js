import React from 'react'
import { connect } from 'react-redux'
import Modal from 'react-modal'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import ExpenseForm from './ExpenseForm'
import RemoveExpenseModal from './RemoveExpenseModal'

export class EditExpensePage extends React.Component {
  state = {
    readyToBeRemoved: undefined,
  }
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/') // throw user to dashboard page
  }

  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id })
    this.props.history.push('/')
  }

  onCloseModal = () => {
    this.setState(() => ({ readyToBeRemoved: undefined }))
  }

  askToRemove = (expense) => {
    this.setState(() => ({ readyToBeRemoved: true }))
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h2 className="page-header__title">Edit Expense</h2>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
          <button className="btn btn--secondary" onClick={this.askToRemove}>
            Remove
          </button>
        </div>
        <RemoveExpenseModal
          readyToBeRemoved={this.state.readyToBeRemoved}
          closeModal={this.onCloseModal}
          expense={this.props.expense}
          onRemove={this.onRemove}
        />
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id),
})
const mapDispatchToProps = dispatch => ({
  startEditExpense: (id, changes) => dispatch(startEditExpense(id, changes)),
  startRemoveExpense: id => dispatch(startRemoveExpense(id)),
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)
