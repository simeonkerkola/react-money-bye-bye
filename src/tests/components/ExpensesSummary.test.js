import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should render ExpensesSummary w/ 1 expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={33} />,
  )

  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary w/ multiple expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={3} expensesTotal={331} />,
  )

  expect(wrapper).toMatchSnapshot()
})
