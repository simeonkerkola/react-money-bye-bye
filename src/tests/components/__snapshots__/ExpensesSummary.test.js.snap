// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary w/ 1 expense 1`] = `
<div>
  <h1>
    1
     
    expense
     totalling 
    33
    €
  </h1>
</div>
`;

exports[`should render ExpensesSummary w/ multiple expenses 1`] = `
<div>
  <h1>
    3
     
    expenses
     totalling 
    331
    €
  </h1>
</div>
`;
